homeassistant/components/tellduslive/__init__.py

```diff
@@ -4,250 +4,174 @@
-            hass.add_job(request_configuration)
-            return
-
-        _LOGGER.debug('Tellstick does support local API')
-
-        # Ignore any known devices
-        if conf and host in conf:
-            _LOGGER.debug('Discovered already known device: %s', host)
-            return
-
-        # Offer configuration of both live and local API
-        request_configuration()
-        request_configuration(host)
-
-    discovery.listen(hass, SERVICE_TELLDUSLIVE, tellstick_discovered)
-
-    if session:
-        _LOGGER.debug('Continuing setup configured by configurator')
-    elif conf and CONF_HOST in next(iter(conf.values())):
-        #  For now, only one local device is supported
-        _LOGGER.debug('Using Local API pre-configured by configurator')
-        session = Session(**next(iter(conf.values())))
-    elif DOMAIN in conf:
-        _LOGGER.debug('Using TelldusLive cloud service '
-                      'pre-configured by configurator')
-        session = Session(PUBLIC_KEY, NOT_SO_PRIVATE_KEY,
-                          application=APPLICATION_NAME, **conf[DOMAIN])
-    elif config.get(DOMAIN):
-        _LOGGER.info('Found entry in configuration.yaml. '
-                     'Requesting TelldusLive cloud service configuration')
-        request_configuration()
-
-        if CONF_HOST in config.get(DOMAIN, {}):
-            _LOGGER.info('Found TelldusLive host entry in configuration.yaml. '
-                         'Requesting Telldus Local API configuration')
-            request_configuration(config.get(DOMAIN).get(CONF_HOST))
 
+async def async_setup(hass, config):
+    """Set up the Telldus Live component."""
+    if DOMAIN not in config:
         return True
-    else:
-        _LOGGER.info('Tellstick discovered, awaiting discovery callback')
-        return True
-
-    if not session.is_authorized:
-        _LOGGER.error('Authentication Error')
-        return False
 
-    client = TelldusLiveClient(hass, config, session)
-    hass.data[DOMAIN] = client
-    client.update()
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={'source': config_entries.SOURCE_IMPORT},
+            data={
+                KEY_HOST: config[DOMAIN].get(CONF_HOST),
+                KEY_SCAN_INTERVAL: config[DOMAIN].get(CONF_UPDATE_INTERVAL),
+            }))
+    return True
 
-    interval = config.get(DOMAIN, {}).get(CONF_UPDATE_INTERVAL,
-                                          DEFAULT_UPDATE_INTERVAL)
-    _LOGGER.debug('Update interval %s', interval)
-    track_time_interval(hass, client.update, interval)
 
+async def async_unload_entry(hass, config_entry):
+    """Unload a config entry."""
+    interval_tracker = hass.data.pop(INTERVAL_TRACKER)
+    interval_tracker()
+    await asyncio.wait([
+        hass.config_entries.async_forward_entry_unload(config_entry, component)
+        for component in hass.data.pop(CONFIG_ENTRY_IS_SETUP)
```

Comment:
We should forward unload to each component, ie each domain that we have set up for the platforms. So that's `'binary_sensor' `, `'cover' `, etc.

Reply:
`CONFIG_ENTRY_IS_SETUP` is a `set()` with the platforms needed (remember that we dynamically load the platform based on what type we discover during update https://github.com/home-assistant/home-assistant/pull/18758/files#diff-6182200c1051279a736e0dd1a80942d9R152)..

Reply:
Yep, but when unloading we need to use the component names.

Reply:
Sorry my bad, it is not what I thought it was..

Reply:
Now it is :).